include/shark/Data/Dataset.h

```diff
@@ -119,9 +119,9 @@ class Data : public ISerializable
 	typedef Type element_type;
 	typedef typename Batch<Type>::shape_type shape_type;
 
-	typedef typename Container::BatchType batch_type;
```

Comment:
If you happen to have a refactoring tool that can replace typedef X Y; with using X = Y; then please, please apply it to the whole library.